cranelift/filetests/filetests/isa/riscv64/umax.clif

```diff
@@ -87,12 +84,15 @@ block0(v0: i64, v1: i64):
 
 ; VCode:
 ; block0:
-;   select a0,a0,a1##condition=(a0 ugt a1)
+;   mv a4,a0
+;   select a0,a4,a1##condition=(a4 ugt a1)
 ;   ret
 ;
 ; Disassembled:
 ; block0: ; offset 0x0
-;   bltu a1, a0, 8
+;   mv a4, a0
+;   mv a0, a4
```

Comment:
We now sometimes get this funky output from regalloc. I think I understand why it's doing this, but maybe there's some way to avoid this?

---

As a side note, maybe we could force the destination register to always be the same as one of the inputs. We would always emit the 2 instruction select and let regalloc emit the move beforehand if it finds it beneficial.

I'm not sure if this is better or not.